src/pages/bridge/Issue.tsx

```diff
@@ -245,18 +247,29 @@ function Issue(props: IssueProps): JSX.Element {
     },
   });
 
-  console.log('walletaccount', walletAccount);
-
   // We watch the amount because we need to re-render the FeeBox constantly
   const amount = watch('amount');
-  const vaults = getVaults();
+
+  useEffect(() => {
+    let combinedVaults: ExtendedRegistryVault[] = [];
+    getVaultsWithIssuableTokens().then((vaultsWithIssuableTokens) => {
+      getVaults().forEach((vaultFromRegistry) => {
+        const found = vaultsWithIssuableTokens?.find(([id, _]) => id.accountId.eq(vaultFromRegistry.id.accountId));
```

Comment:
Since vaults are only uniquely identified with the `accountId` + `currencyPair` we need to also check if the currency pairs are equal in this `find` statement. Otherwise we might see very weird side-effects once a vault registers the same `accountId` for multiple currency pairs.

Reply:
You are right, somehow I thought I was comparing the whole thing. I fixed it now.